cli/command/engine/update.go

```diff
@@ -0,0 +1,66 @@
+package engine
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/containerizedengine"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+type updateOptions struct {
+	version        string
+	registryPrefix string
+	image          string
+}
+
+func newUpdateCommand(dockerCli command.Cli) *cobra.Command {
+	options := updateOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "update [OPTIONS]",
+		Short: "Update a local engine",
+		Args:  cli.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runUpdate(dockerCli, options)
+		},
+	}
+	flags := cmd.Flags()
+
+	flags.StringVar(&options.version, "version", "", "Specify engine version")
+	flags.StringVar(&options.image, "engine-image", "", "Specify engine image")
+	flags.StringVar(&options.registryPrefix, "registry-prefix", "docker.io/docker", "Override the default location where engine images are pulled")
+
+	return cmd
+}
+
+func runUpdate(dockerCli command.Cli, options updateOptions) error {
+	ctx := context.Background()
+	client, err := dockerCli.NewContainerizedEngineClient()
+	if err != nil {
+		return errors.Wrap(err, "unable to access local containerd")
+	}
+	defer client.Close()
+	if options.image == "" {
+		currentEngine, _, err := client.GetCurrentEngineVersion(ctx)
+		if err != nil {
+			return err
+		}
+		options.image = currentEngine
+	}
+	opts := containerizedengine.EngineInitOptions{
+		RegistryPrefix: options.registryPrefix,
+		EngineImage:    options.image,
+		EngineVersion:  options.version,
+	}
+
+	err = client.DoUpdate(ctx, opts, dockerCli.Out(), authResolver(dockerCli))
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(dockerCli.Out(), "Success!  The docker engine is now running.\n")
```

Comment:
nit: fmt.Fprintln(dockerCli.Out(), "Success!  The docker engine is now running.")